### DISCUSSION TITLE
Updating price does not work - python

### DISCUSSION THREAD

**Discussion by stackiflowi** [etsy/open-api#974] [Q&A]
Hi,

I implemented the getListingInventory API call which works fine. Also, i implemented the updateListingInventory, however, that seems to have some problem, and I can't figure out what is that.

How I'm trying to test is that I took the result of the getListingInventory, I change a price value of it, and I'm trying to update the real etsy product, however, it has no affect.

My token has all the scopes.

I'm using the following command for update:

requests.put('https://openapi.etsy.com/v3/application/listings/111111111111111/inventory', headers=header, data=update_dict)

The status code is 200, however, in the response the value of the price is the original one, not the changed one, and also it's not reflected on Etsy.

Is there any problem with my workflow?

Thanks.

**Comment by stackiflowi** [comment]
Anyone would have a command on this?

I'd appreciate some working examples as well which takes basically the result of getListingInventory, modifies it, and puts it back.



Thanks.

**Reply by muratali016** [reply]
If you are using Python I advise you to use the etsyv3 library which makes a lot of things easier for you. You can change the price by using this: https://github.com/anitabyte/etsyv3/issues/2#issuecomment-1528871982 from the library. The library doesn't do anything but send requests in the background:

**Reply by stackiflowi** [reply]
@muratali016 Thanks. I tried it out, but as I thought, I got the same behaviour. I got back a 200 answer, but nothing is getting changed.

In my listing, I've 98 products. I'm using the getListingInventory method to fetch these, I change something in it, and I PUT it back via the updateListingInventory, but nothing is changed on Etsy side, and even I got back 200 as status_code, the returned json contains only one element from the 98. It has a - to me - unknown product_id.

Could anyone send me a working payload for this API call?

**Reply by ghost** [reply]
@stackiflowi mail me at: [murataliavcu1@gmail.com](mailto:murataliavcu1@gmail.com) I can help you out

**Reply by Gareth064** [reply]
@stackiflowi if you are sending a payload to the API and getting a 200 back, but not seeing the changes on the GUI then sounds like a bug which Etsy will help you with. Email then at developer@etsy.com

As for the unknown product_id. Pretty sure the product_id is a "point in time" id, and changes every time you make a "change" to the product/offering/listing. If you want a consistent id to work with a track then you need to add your own SKU's to the listing/variations.

**Reply by Gareth064** [reply]
have a look at this discussion also, might have some pointers in it https://github.com/etsy/open-api/discussions/691

**Reply by stackiflowi** [reply]
@muratali16 Thanks, I sent You the email.
@Gareth064 I also sent it to developer@etsy.com

I exactly made the same as @etsyachristensen suggested in https://github.com/etsy/open-api/discussions/691 .

**Reply by muratali016** [reply]
@stackiflowi problem solved

**Reply by Gareth064** [reply]
Wanna share what the problem was and how it was resolved?

**Reply by muratali016** [reply]
@Gareth064 We used this method: https://github.com/anitabyte/etsyv3/issues/2#issuecomment-1528871982
and this time it worked

**Reply by Gareth064** [reply]
Cheers. This is very specific to python obviously. @stackiflowi if you get a response from the Etsy Devs about why you were getting the 200 response code back from the API, but the price wasn't updating, then please still let us know here what they say.

**Reply by muratali016** [reply]
@Gareth064 There are multiple price elements because of the different SKUs of the particular product in the response from etsy api. Probably he wasn't changing all of it.

**Reply by muratali016** [reply]
Please check the response "get_listing_inventory" that's a only one products response
`{'products': [{'product_id': 8085849301, 'sku': '', 'is_deleted': False, 'offerings': [{'offering_id': 8320940184, 'quantity': 867, 'is_enabled': True, 'is_deleted': False, 'price': {'amount': 800, 'divisor': 100, 'currency_code': 'USD'}}], 'property_values': [{'property_id': 513, 'property_name': 'Leather Color', 'scale_id': None, 'scale_name': None, 'value_ids': [50135267836], 'values': ['Black']}, {'property_id': 514, 'property_name': 'Foil Option', 'scale_id': None, 'scale_name': None, 'value_ids': [98686866789], 'values': ['No Personalization']}]}, {'product_id': 8322870934, 'sku': '', 'is_deleted': False, 'offerings': [{'offering_id': 8166948977, 'quantity': 984, 'is_enabled': True, 'is_deleted': False, 'price': {'amount': 1600, 'divisor': 100, 'currency_code': 'USD'}}], 'property_values': [{'property_id': 513, 'property_name': 'Leather Color', 'scale_id': None, 'scale_name': None, 'value_ids': [50135267836], 'values': ['Black']}, {'property_id': 514, 'property_name': 'Foil Option', 'scale_id': None, 'scale_name': None, 'value_ids': [104787377652], 'values': ['Add Initials']}]}, {'product_id': 8322870938, 'sku': '', 'is_deleted': False, 'offerings': [{'offering_id': 8320940190, 'quantity': 0, 'is_enabled': True, 'is_deleted': False, 'price': {'amount': 800, 'divisor': 100, 'currency_code': 'USD'}}], 'property_values': [{'property_id': 513, 'property_name': 'Leather Color', 'scale_id': None, 'scale_name': None, 'value_ids': [53557181503], 'values': ['Blue']}, {'property_id': 514, 'property_name': 'Foil Option', 'scale_id': None, 'scale_name': None, 'value_ids': [98686866789], 'values': ['No Personalization']}]}, {'product_id': 8085849309, 'sku': '', 'is_deleted': False, 'offerings': [{'offering_id': 8166948981, 'quantity': 0, 'is_enabled': True, 'is_deleted': False, 'price': {'amount': 1600, 'divisor': 100, 'currency_code': 'USD'}}], 'property_values': [{'property_id': 513, 'property_name': 'Leather Color', 'scale_id': None, 'scale_name': None, 'value_ids': [53557181503], 'values': ['Blue']}, {'property_id': 514, 'property_name': 'Foil Option', 'scale_id': None, 'scale_name': None, 'value_ids': [104787377652], 'values': ['Add Initials']}]}, {'product_id': 10860998587, 'sku': '', 'is_deleted': False, 'offerings': [{'offering_id': 10967035339, 'quantity': 923, 'is_enabled': True, 'is_deleted': False, 'price': {'amount': 800, 'divisor': 100, 'currency_code': 'USD'}}], 'property_values': [{'property_id': 513, 'property_name': 'Leather Color', 'scale_id': None, 'scale_name': None, 'value_ids': [64313028245], 'values': ['Light Brown']}, {'property_id': 514, 'property_name': 'Foil Option', 'scale_id': None, 'scale_name': None, 'value_ids': [98686866789], 'values': ['No Personalization']}]}, {'product_id': 11178202560, 'sku': '', 'is_deleted': False, 'offerings': [{'offering_id': 10967035343, 'quantity': 992, 'is_enabled': True, 'is_deleted': False, 'price': {'amount': 1600, 'divisor': 100, 'currency_code': 'USD'}}], 'property_values': [{'property_id': 513, 'property_name': 'Leather Color', 'scale_id': None, 'scale_name': None, 'value_ids': [64313028245], 'values': ['Light Brown']}, {'property_id': 514, 'property_name': 'Foil Option', 'scale_id': None, 'scale_name': None, 'value_ids': [104787377652], 'values': ['Add Initials']}]}, {'product_id': 10860998591, 'sku': '', 'is_deleted': False, 'offerings': [{'offering_id': 10967035345, 'quantity': 902, 'is_enabled': True, 'is_deleted': False, 'price': {'amount': 800, 'divisor': 100, 'currency_code': 'USD'}}], 'property_values': [{'property_id': 513, 'property_name': 'Leather Color', 'scale_id': None, 'scale_name': None, 'value_ids': [55178982649], 'values': ['Dark Brown']}, {'property_id': 514, 'property_name': 'Foil Option', 'scale_id': None, 'scale_name': None, 'value_ids': [98686866789], 'values': ['No Personalization']}]}, {'product_id': 11178202562, 'sku': '', 'is_deleted': False, 'offerings': [{'offering_id': 10967035347, 'quantity': 978, 'is_enabled': True, 'is_deleted': False, 'price': {'amount': 1600, 'divisor': 100, 'currency_code': 'USD'}}], 'property_values': [{'property_id': 513, 'property_name': 'Leather Color', 'scale_id': None, 'scale_name': None, 'value_ids': [55178982649], 'values': ['Dark Brown']}, {'property_id': 514, 'property_name': 'Foil Option', 'scale_id': None, 'scale_name': None, 'value_ids': [104787377652], 'values': ['Add Initials']}]}, {'product_id': 8085849313, 'sku': '', 'is_deleted': False, 'offerings': [{'offering_id': 8166948989, 'quantity': 0, 'is_enabled': True, 'is_deleted': False, 'price': {'amount': 800, 'divisor': 100, 'currency_code': 'USD'}}], 'property_values': [{'property_id': 513, 'property_name': 'Leather Color', 'scale_id': None, 'scale_name': None, 'value_ids': [52203504701], 'values': ['Cream']}, {'property_id': 514, 'property_name': 'Foil Option', 'scale_id': None, 'scale_name': None, 'value_ids': [98686866789], 'values': ['No Personalization']}]}, {'product_id': 8085849319, 'sku': '', 'is_deleted': False, 'offerings': [{'offering_id': 8166948995, 'quantity': 0, 'is_enabled': True, 'is_deleted': False, 'price': {'amount': 1600, 'divisor': 100, 'currency_code': 'USD'}}], 'property_values': [{'property_id': 513, 'property_name': 'Leather Color', 'scale_id': None, 'scale_name': None, 'value_ids': [52203504701], 'values': ['Cream']}, {'property_id': 514, 'property_name': 'Foil Option', 'scale_id': None, 'scale_name': None, 'value_ids': [104787377652], 'values': ['Add Initials']}]}, {'product_id': 8085849325, 'sku': '', 'is_deleted': False, 'offerings': [{'offering_id': 8166949001, 'quantity': 0, 'is_enabled': True, 'is_deleted': False, 'price': {'amount': 700, 'divisor': 100, 'currency_code': 'USD'}}], 'property_values': [{'property_id': 513, 'property_name': 'Leather Color', 'scale_id': None, 'scale_name': None, 'value_ids': [53315356698], 'values': ['Gray']}, {'property_id': 514, 'property_name': 'Foil Option', 'scale_id': None, 'scale_name': None, 'value_ids': [98686866789], 'values': ['No Personalization']}]}, {'product_id': 8322870948, 'sku': '', 'is_deleted': False, 'offerings': [{'offering_id': 8166949005, 'quantity': 0, 'is_enabled': True, 'is_deleted': False, 'price': {'amount': 1600, 'divisor': 100, 'currency_code': 'USD'}}], 'property_values': [{'property_id': 513, 'property_name': 'Leather Color', 'scale_id': None, 'scale_name': None, 'value_ids': [53315356698], 'values': ['Gray']}, {'property_id': 514, 'property_name': 'Foil Option', 'scale_id': None, 'scale_name': None, 'value_ids': [104787377652], 'values': ['Add Initials']}]}], 'price_on_property': [513, 514], 'quantity_on_property': [513, 514], 'sku_on_property': [513, 514], 'listing': None}`

**Reply by stackiflowi** [reply]
@Gareth064 The problem was that I didn't delete the scale_name from the received inventory, and that caused the problem. However, it's a mistery why I still received 200. If I do not delete for instance the is_deleted, I got 400 with detailed error message.

I did not receive any answer from Etsy yet.

**Reply by stackiflowi** [reply]
@Gareth064 The problem was an unfortunate type error. I deleted scala_name instead of scale_name from the dictionary, and since I didn't receive error message, I couldn't figure it out... .
